pkg/controller/node/nodecontroller.go

```diff
@@ -145,38 +150,43 @@ func (nc *NodeController) Run(period time.Duration) {
 	}, nc.nodeMonitorPeriod)
 
 	go util.Forever(func() {
-		nc.podEvictor.TryEvict(func(nodeName string) { nc.deletePods(nodeName) })
+		nc.podEvictor.Try(func(value TimedValue) (bool, time.Duration) {
```

Comment:
The queue was intended to be generic enough to reuse (it's useful for any
client code that wants to wait X time before retrying an item again, like
controllers if a retry is needed), could be an interface{} but that was
more changes.

On Aug 24, 2015, at 9:58 AM, Marek Grabowski <notifications@github.com>
wrote:

In pkg/controller/node/nodecontroller.go
<https://github.com/kubernetes/kubernetes/pull/12962#discussion_r37747669>:

> @@ -145,38 +150,43 @@ func (nc *NodeController) Run(period time.Duration) {
>  	}, nc.nodeMonitorPeriod)
>
>  	go util.Forever(func() {
> -		nc.podEvictor.TryEvict(func(nodeName string) { nc.deletePods(nodeName) })
> +		nc.podEvictor.Try(func(value TimedValue) (bool, time.Duration) {

I'd call variable timedNodeName, or something else more descriptive than
plain 'value'

—
Reply to this email directly or view it on GitHub
<https://github.com/kubernetes/kubernetes/pull/12962/files#r37747669>.